www/src/components/homepage/homepage-ecosystem.js

```diff
@@ -0,0 +1,226 @@
+import { PluginsIcon, StartersIcon } from "../../assets/ecosystem-icons"
+
+import { rhythm, options } from "../../utils/typography"
+import presets, { colors } from "../../utils/presets"
+
+import { SCROLLER_CLASSNAME } from "../../utils/scrollers-observer"
+
+const Sections = styled(`div`)`
+  display: flex;
+  flex-direction: column;
+
+  ${presets.Tablet} {
+    flex-direction: row;
+    margin: 0 -8px;
+  }
+
+  ${presets.Desktop} {
+    margin: 0 1.5rem 0 2.5rem;
+  }
+`
+
+const Section = styled(EcosystemSection)`
+  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.2);
+  border-radius: ${presets.radiusLg}px;
+  margin-bottom: ${rhythm(presets.gutters.default / 2)};
+  padding: ${rhythm(options.blockMarginBottom)};
+
+  ${presets.Tablet} {
+    padding: ${rhythm(options.blockMarginBottom)};
+    margin: 0 8px 0px;
+
+    :last-child {
+      align-self: stretch;
+    }
+  }
+`
+
+const SubTitle = styled(`h3`)`
+  color: ${colors.lemon};
+  font-size: 1.2rem;
+  margin-top: 2rem;
+
+  ${presets.Tablet} {
+    margin-left: 3rem;
+  }
+
+  ${presets.Desktop} {
+    margin-left: 6rem;
+  }
+`
+
+const FeaturedItems = styled(EcosystemFeaturedItemsRootBase)`
+  margin: 0 -${rhythm(presets.gutters.default / 2)};
+
+  ${presets.Desktop} {
+    margin: 0;
+    margin-left: calc(3rem - (${rhythm(options.blockMarginBottom)}));
+    margin-right: 1rem;
+    overflow-x: auto;
+  }
+
+  ${presets.Hd} {
+    margin-right: 3rem;
+  }
+`
+
+const FeaturedItemsList = styled(EcosystemFeaturedItemsListBase)`
+  width: ${props => `calc(320px * ${props.numberOfItems})`};
+  padding: 0 calc(${rhythm(options.blockMarginBottom)} - 7px) 0;
+
+  ${presets.Desktop} {
+    width: 100%;
+    margin: 0;
+    flex-wrap: wrap;
+  }
+`
+
+const FeaturedItem = styled(EcosystemFeaturedItem)`
+  margin: 0 6px 6px 0;
+
+  ${presets.Tablet} {
+    border-bottom: none;
+    padding: 5px;
+  }
+
+  ${presets.Desktop} {
+    flex-basis: 30%;
+
+    :nth-child(4) {
+      margin-left: 8%;
+    }
+  }
+
+  ${FeaturedItemBlockLink} {
+    padding-left: calc(${rhythm(3 / 4)} + 1.1rem);
+    position: relative;
+
+    /* this ovveride the .main-body a style*/
+    font-weight: normal;
+    box-shadow: none;
+
+    ${presets.Tablet} {
+      border-radius: ${presets.radiusLg}px;
+    }
+
+    ${presets.Desktop} {
+      :hover {
+        background: ${colors.ui.whisper};
+      }
+    }
+
+    :before {
+      background: ${props =>
+        props.item.type === `Starter` ? `#dcfffd` : `#ffeccd`};
+      border-radius: ${presets.radiusLg}px 0 0 ${presets.radiusLg}px;
+      bottom: 0;
+      content: "";
+      left: 0;
+      position: absolute;
+      top: 0;
+      width: 1.1rem;
+    }
+
+    :after {
+      bottom: 0;
+      content: "${props => props.item.type}";
+      color: ${props =>
+        props.item.type === `Starter` ? ` #0a75c2` : `#9e6100`};
+      font-family: ${options.headerFontFamily.join(`,`)};
+      font-size: 0.8rem;
+      left: 0;
```

Comment:
Total nit, but I find this hard to follow. I usually group properties related to positioning, e.g.

```css
position: absolute;
bottom: 0;
left: 0;

/* Rest of properties */
```

Reply:
I usually order properties alphabetically, but I can adjust

Reply:
Just saw this in the mail and thought "Greg maybe had been following what I've been doing on `www`". I also order alphabetically…when I think of it that is 😉(VS Code plugins doing that which I checked have a hard time with multiple lines).

I still often find myself keeping `position`-related stuff together initially, and when I get to reorder them I still often shrug, but in general I find alphabetic ordering to have it's advantages, and an OK "least common denominator". 🤐 😸

Reply:
> I find this hard to follow.

Def. is!

Reply:
I always go alphabetical, too. I think there's a good argument for contextual grouping, too, but that assumes we all think of the same rules in the same context. 🤷‍♂️